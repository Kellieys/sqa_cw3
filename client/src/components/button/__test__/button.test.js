import React from 'react';
import ReactDom from 'react-dom';
import Button from '../button';

import { render, cleanup } from "@testing-library/react";
import "jest-dom/extend-expect";

import renderer from "react-test-renderer";

afterEach(cleanup);

it("renders without crashing" , () => {
    const div = document.createElement("div");
    ReactDom.render(<Button />, div);
});

it("renders button correctly", () => {
    const { getByTestId } = render(<Button label="hello" />);
    expect(getByTestId('button')).toHaveTextConetnt("hello")
});

it("matches snapshot", () => {
    const tree = renderer.create(<Button label="hello"></Button>).toJSON();
    expect(tree).toMatchSnapshot();
});