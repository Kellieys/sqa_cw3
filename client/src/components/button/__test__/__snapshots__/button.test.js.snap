// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<button
  className="button-search"
  data-testid="button"
  type="submit"
>
  Search
</button>
`;
